packages/appstore/src/components/main-title-bar/__tests__/asset-summary.spec.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import AssetSummary from '../asset-summary';
+import { render, screen } from '@testing-library/react';
+import { StoreProvider, mockStore } from '@deriv/stores';
+import { isMobile } from '@deriv/shared';
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isMobile: jest.fn(() => false),
+}));
+
+jest.mock('Components/pre-loader/total-assets-loader', () => ({
+    __esModule: true,
+    default: () => <div>TotalAssetLoader</div>,
+}));
+
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    Popover: jest.fn(props => (
+        <>
+            <span>{props.message}</span>
+            <span>{props.alignment}</span>
+        </>
+    )),
+}));
+
+describe('AssetSummary', () => {
+    it('should render asset summary', () => {
+        const mock = mockStore({});
+
+        const wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock}>{children}</StoreProvider>
+        );
+
+        const { container } = render(<AssetSummary />, {
+            wrapper,
+        });
+        expect(container).toBeInTheDocument();
+    });
+
+    it('should render asset summary with total asset', () => {
+        const mock = mockStore({});
+
+        const wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock}>{children}</StoreProvider>
+        );
+
+        const { container } = render(<AssetSummary />, {
+            wrapper,
+        });
+        expect(container).toBeInTheDocument();
+        expect(screen.getByText('Total asset')).toBeInTheDocument();
+    });
+
+    it('should not render the total asset text if isMobile is true', () => {
+        const mock = mockStore({});
+
+        const wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock}>{children}</StoreProvider>
+        );
+        isMobile.mockImplementation(() => true);
+        const { container } = render(<AssetSummary />, {
+            wrapper,
+        });
+        expect(container).toBeInTheDocument();
```

Comment:
I see a lot of repeated values checking for AssetSummary. can we make it into a test function and call it inside the test case?